app/serializers/search_and_compare/course_serializer.rb

```diff
@@ -63,5 +62,53 @@ def get_subjects
         }
       end
     end
+
+    # Campuses_related_Mapping
+    attribute(:Campuses)                              { get_campuses }
+    # using server time not utc, so it's local time?
+    attribute(:ApplicationsAcceptedFrom)              { object.applications_open_from.to_date.strftime('%Y-%m-%dT%H:%M:%S') }
+    attribute(:HasVacancies)                          { object.has_vacancies? }
+    def get_campus_default_value
+      {
+        Id: 0,
+        LocationId: nil,
+        Course: nil,
+      }
+    end
+
+    def get_location_default_value
+      {
+        Id: 0,
+        FormattedAddress: nil,
+        GeoAddress: nil,
+        Latitude: nil,
+        Longitude: nil,
+        LastGeocodedUtc: '0001-01-01T00:00:00'
+      }
+    end
+
+    def get_address(address1:, address2:, address3:, address4:, postcode:)
```

Comment:
```
def full_address
    [address1, address2, address3, address4, postcode].select(&:present?).join("<br> ").html_safe
end
```

?

Reply:
cannot use html related bits as it needs to be feed in to geocoder

`reject` is always better then `select`

`present?` is that true for `""`?

Reply:
renamed method to `full_address`

its going to be needed for provider location and contact details address

Reply:
Why is `reject` better than `select`?

Reply:
just a passing motion, 

select is opposite to reject
present is opposite to blank

ruby is about the only one that has the opposite named method to everything


> array reject blank join

> array select present join

either one is fine by me, favouring `reject` i can only why....

Reply:
wat?

Reply:
> > array reject blank join

```array.reject(&:blank?).join('/n')```
 
> > array select present join

```array.select(&:present?).join("<br> ") ```